app/services/hyrax/valkyrie_upload.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module Hyrax::ValkyrieUpload
+  # @param [IO] io
+  # @param [String] filename
+  # @param [Hyrax::FileSet] file_set
+  # @param [RDF::URI] use
+  # @see Hyrax::FileMetadata::Use
+  # @return [Hyrax::FileMetadata] the metadata representing the uploaded file
+  def self.file(io:, filename:, file_set:, use: Hyrax::FileMetadata::Use::ORIGINAL_FILE, storage_adapter: Hyrax.storage_adapter) # rubocop:disable Metrics/MethodLength
+    file_metadata = Hyrax.persister.save(
```

Comment:
Here's where Wings gets tripped up; this `#save` needs to happen so that the `FileMetadata` has an ID for the `#upload` below, but Wings doesn't want to persist metadata without all its parent objects being persisted.

Reply:
https://app.circleci.com/pipelines/github/samvera/hyrax/7613/workflows/d3883e13-1d8a-4799-83bc-d0e04067552d/jobs/60631